modules/imgproc/src/pyramids.cpp

```diff
@@ -492,10 +471,23 @@ template<> int PyrDownVecV<int, uchar>(int** src, uchar* dst, int width)
         r2 = v_load(row2 + x);
         r3 = v_load(row3 + x);
         r4 = v_load(row4 + x);
-        t0 = r0 + r4 + (r2 + r2) + ((r1 + r3 + r2) << 2);
+        t0 = v_add(v_add(v_add(r0, r4), v_add(r2, r2)), v_shl<2>(v_add(v_add(r1, r3), r2)));
 
         *((unaligned_int*) (dst + x)) = v_reinterpret_as_s32(v_rshr_pack<8>(v_pack_u(t0, t0), v_setzero_u16())).get0();
     }
+    #else
+    for (; x <= width - 1; x += 1)
```

Comment:
The `CV_SIMD128` code block does not seem to encounter the tails, because the data is 128-bit aligned, as follows:
1. In the `CV_SIMD128` code block, for `row0~row4`, each loop loads and processes 128-bit data.
2. The `row0~row4` are `src[0]~src[4]` respectively, coming from the parameter `int** src`
3. For the call to the `PyrDownVecV` function (only one), `WT* rows[PD_SZ]` is passed in as argument.
4. The size of `rows[0]~rows[4]` is determined by `buf` and `bufstep`, and they are 16byte aligned, that is, 128-bit.
```
int bufstep = (int)alignSize(dsize.width*cn, 16); //line 928
WT* buf = alignPtr((WT*)_buf.data(), 16); //line 930
```
If the above analysis is correct, then I think the plain loop should only work on non-CV_SIMD128 cases, that is, in `#else`

BTW, this part of the code has been working like this for many years, which may indirectly means that the original implementation (128-fixed related code) does not need to consider the tail.